### ISSUE TITLE
False positive: inherit-non-class on subclass of metaclass-constructed class

### ISSUE THREAD

**Issue by federicobond** [microsoft/python-language-server#1764]
## Environment data

- Language Server version: 0.4.71.0 
- Python version (& distribution if applicable, e.g. Anaconda): Python 3.7

## Expected behaviour

No errors.

## Actual behaviour

PLS reports `inherit-non-class` in the declaration of Subclass. See code snippet below.

## Code Snippet / Additional lnformation

```python
class MyMetaclass(type):
    pass

MyClass = MyMetaclass('MyClass', (), {})

class Subclass(MyClass):
    pass
```

**Comment by MikhailArkhipov** [comment]
Most probably it perceives `MyClass` as instance.

**Comment by federicobond** [comment]
Yes, it should add a case to consider a class any instance of a `type`.

**Comment by MikhailArkhipov** [comment]
Related https://github.com/microsoft/python-language-server/issues/1367

**Comment by federicobond** [comment]
Thank you! ❤️

**Comment by jakebailey** [comment]
Fixed in 0.4.101+, which is in beta (but will be released to everyone in stable in the near future).
